src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java

```diff
@@ -224,23 +232,51 @@ private static void validateTableAccess(Client engineClient, JSONObject mapping,
         
         String database = mapping.getString("db");
         String table = mapping.getString("table");
+        String format = mapping.optString("format");
+        String mappingName = mapping.optString("mapping");
+        boolean hasAccess = false;
         try {
+            try {
+                KustoOperationResult rs = engineClient.execute(database, String.format(FETCH_TABLE_QUERY, table));
+                if ((int) rs.getPrimaryResults().getData().get(0).get(0) >= 0) {
+                    hasAccess = true;
+                }
+
+            } catch (DataServiceException e) {
+                hasAccess = false;
+                databaseTableErrorList.add(String.format("Database:%s Table:%s | table not found", database, table));
+            }
+            if(hasAccess) {
+                try {
+                    KustoOperationResult rp = engineClient.execute(database, String.format(FETCH_TABLE_MAPPING_QUERY, table, format, mappingName));
+                    if (rp.getPrimaryResults().getData().get(0).get(0).toString().equals(mappingName)) {
+                        hasAccess = true;
```

Comment:
@ohadbitt The query that we currently using will result in only a single record if the mapping exists, else it will throw a mapping not found exception. The If is just for a more robust validation